### ISSUE TITLE
SC2102 Mistakes IPv6 Address with Range

### ISSUE THREAD

**Issue by gengwg** [koalaman/shellcheck#2450]
#### For bugs
- Rule Id (if any, e.g. SC1000):  SC2102
- My shellcheck version (`shellcheck --version` or "online"):  online
- [ ] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086) https://github.com/koalaman/shellcheck/wiki/SC2102
- [ ] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [ ] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash
curl -k https://[2620:10d:c0a8:2b::127]/ 
```

#### Here's what shellcheck currently says:

```
$ shellcheck myscript
 
[Line 2:](javascript:setPosition(2, 17))
curl -k https://[2620:10d:c0a8:2b::127]/
                ^-- [SC2102](https://github.com/koalaman/shellcheck/wiki/SC2102) (info): Ranges can only match single chars (mentioned due to duplicates).
```

#### Here's what I wanted or expected to see:

It's IPv6. Should be no warning.

**Comment by koalaman** [comment]
The error is certainly misleading, but this construct does require quoting:

```
$ mkdir -p https:/2/foo
$ echo https://[2620:10d:c0a8:2b::127]/
https://2/
```

While a directory named `https:` is unlikely, it will also trigger `nullglob` and `failglob` so it's better to quote it. I've updated the wiki. Thanks!
